treeherder/log_parser/utils.py

```diff
@@ -1,66 +1,47 @@
-import logging
+import os
 
 import simplejson as json
-from requests.exceptions import HTTPError
+from jsonschema import (ValidationError,
+                        validate)
 
-from treeherder.etl.artifact import (serialize_artifact_json_blobs,
-                                     store_job_artifacts)
-from treeherder.log_parser.artifactbuildercollection import (ArtifactBuilderCollection,
-                                                             LogSizeException)
-from treeherder.model.models import JobLog
 
-logger = logging.getLogger(__name__)
+def _lookup_extra_options_max(schema):
+    return (
+        schema
+        ["definitions"]
+        ["suite_schema"]
+        ["properties"]
+        ["extraOptions"]
+        ["items"]
+        ["maxLength"]
+    )
 
 
-def extract_text_log_artifacts(job_log):
-    """Generate a set of artifacts by parsing from the raw text log."""
+with open(os.path.join('schemas', 'performance-artifact.json')) as f:
+    PERFHERDER_SCHEMA = json.load(f)
+    MAX_LENGTH = _lookup_extra_options_max(PERFHERDER_SCHEMA)
+    SECOND_MAX_LENGTH = 45
 
-    # parse a log given its url
-    artifact_bc = ArtifactBuilderCollection(job_log.url)
-    artifact_bc.parse()
 
-    artifact_list = []
-    for name, artifact in artifact_bc.artifacts.items():
-        artifact_list.append({
-            "job_guid": job_log.job.guid,
-            "name": name,
-            "type": 'json',
-            "blob": json.dumps(artifact)
-        })
+def validate_perf_data(performance_data: dict):
+    validate(performance_data, PERFHERDER_SCHEMA)
+    if SECOND_MAX_LENGTH > MAX_LENGTH:
```

Comment:
Let's remove this condition, as we provided test coverage for it.

Reply:
OK